bittensor/utils/subprocess/commit_reveal.py

```diff
@@ -0,0 +1,784 @@
+
+
+def handle_client_connection(client_socket: socket.socket):
+    """
+    Handles incoming client connections for the socket server.
+    Args:
+        client_socket (socket.socket): The client socket connection.
+    """
+    try:
+        while True:
+            request = client_socket.recv(1024).decode()
+            if not request:
+                break
+            if request.startswith("revealed_hash_batch"):
+                try:
+                    command = "revealed_hash_batch"
+                    json_start_index = request.index("[")
+                    json_payload = request[json_start_index:]
+                    args = json.loads(json_payload)
+                    revealed_commit_batch(args)
+                except json.JSONDecodeError as e:
+                    print(f"Error decoding JSON for {command}: {e}")
+                except Exception as e:
+                    print(f"Error processing {command}: {e}")
+            else:
+                args = shlex.split(request)
+                command = args[0]
+                commands = {
+                    "revealed_hash": lambda: revealed_commit(args[1]),
+                    "revealed_hash_batch": lambda: revealed_commit_batch(
+                        json.loads(args[1])
+                    ),
+                    "committed": lambda: committed(
+                        Commit(
+                            wallet_hotkey_name=args[3],
+                            wallet_hotkey_ss58=args[4],
+                            wallet_name=args[1],
+                            wallet_path=args[2],
+                            commit_hash=args[8],
+                            netuid=int(args[9]),
+                            commit_block=int(args[5]),
+                            reveal_block=int(args[6]),
+                            expire_block=int(args[7]),
+                            uids=json.loads(args[10]),
+                            weights=json.loads(args[11]),
+                            salt=json.loads(args[12]),
+                            version_key=int(args[13]),
+                        )
+                    ),
+                    "terminate": lambda: terminate_process(None, None),
+                }
+                if command in commands:
+                    try:
+                        commands[command]()
+                    except (IndexError, ValueError, json.JSONDecodeError) as e:
+                        print(f"Error in processing command {command}: {e}")
+                else:
+                    print(f"Command not recognized: {command}")
+    except socket.error as e:
+        print(f"Socket error: {e}")
+    except Exception as e:
+        print(f"Error: {e}")
+    finally:
+        client_socket.close()
+
+
+def start_socket_server():
+    """
+    Starts the socket server to listen for incoming connections.
+    """
+    server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    server.bind(("127.0.0.1", 9949))
```

Comment:
Have you considered the option when the port is busy for some reason? The user may not know about it.
As an idea: Perhaps it is worth implementing a checking mechanism before starting the process, otherwise inform the user about it.